util/flocks/src/frwlock/read_guard.rs

```diff
@@ -3,20 +3,22 @@ use rustix::{
     fs::{flock, FlockOperation},
     fd::AsFd
 };
+use std::marker::PhantomData;
 
-pub struct FrwLockReadGuard<T: AsFd> {
-    pub(crate) data: *const T
+pub struct FrwLockReadGuard<'a, T: AsFd> {
+    pub(crate) data: *const T,
+    pub(crate) _marker: PhantomData<&'a T>
```

Comment:
I don't see the point of making it this kludgy. If you can guarantee the `'a` lifetime for the object as you claim you do on the type system level, why not just make the `&'a T` borrow the inner value of the guard? If this is for relaxing the bounds on the standard-defined `Sync`/`Send` for `&'a T` (as the custom impls above suggest), the need to do it should be carefully scrutinized, because we may be violating soundness of the pointer access.

The `T` object must own the resource behind the file descriptor. If it's allowed to drop before the lock object, the file descriptor may be reclaimed by another object, e.g. a different file under advisory locking in our system, so dropping the lock guard will call `flock` on that.

Reply:
As already suggested in https://github.com/movementlabsxyz/movement/pull/137#discussion_r1652463085,
`BorrowedFd` might be the solution we need here.

Reply:
`BorrowedFd` sounds like it would make sense.